### ISSUE TITLE
gist:isAffectedBy and gist:isTriggeredBy

### ISSUE THREAD

**Issue by justin2004** [semanticarts/gist#741]
Shouldn't `gist:isTriggeredBy` be a subproperty `gist:isAffectedBy`?

**Comment by rjyounes** [comment]
Makes sense to me.

**Comment by justin2004** [comment]
Also I think perhaps `gist:governs` should be a subproperty of `gist:affects` (the inverse of `gist:isAffectedBy`).

**Comment by justin2004** [comment]
Also `gist:isUnderJurisdictionOf` might be a subproperty of `gist:isAffectedBy`.

**Comment by justin2004** [comment]
Also `gist:produces` seems like a sub prop of `gist:affects`...
but that makes me worry that `gist:affects` and `gist:isAffectedBy` should be at the root of the obj prop tree and nearly all other obj props should be subprops of them.

**Comment by uscholdm** [comment]
> hat makes me worry that `gist:affects` and `gist:isAffectedBy` should be at the root of the obj prop tree and nearly all other obj props should be subprops of them.

Worry not, an identifier does not affect what it identifies. Id guess 30-70% of properties are not subprops of gist:affects.

**Comment by uscholdm** [comment]
> Also `gist:produces` seems like a sub prop of `gist:affects`...

Maybe, but i don't think so - it a produce affects existence, in a weird way, but something that does not exist cannot be affected.

**Comment by rjyounes** [comment]
@justin2004 I had the same concern (in which case we wouldn't need those predicates at all) but I scanned our list of predicates and found many that wouldn't fall under the notion of 'affects.' @uscholdm has pointed out one, and there are others:

```
comesFromPlcae
containsGeographically
follows
precedes
hasAddress
hasAltitude
hasDenominator
hasMagnitude
```
... and many more.

Nevertheless, I'd say `affects` is somewhat vacuous. @uscholdm Do you know of cases where we'd want to use it rather than one of the more specific properties? If not we might want to remove it.

**Comment by uscholdm** [comment]
Phil is using it for tasks, which often are there to affect a specific thing.  The task to merge gist's PR123 affects gist.

**Comment by rjyounes** [comment]
Could we say in that case it contributes to gist? Or are we distinguishing the task of merging from the merge itself? I.e., the task affects gist, the merge contributes to gist. That's a pretty subtle distinction - I doubt if there's a need for it.

**Comment by uscholdm** [comment]
There is some history here.  Long ago, gist had 4 top level proporties and pretty much all the other properties were subproperities of one of the 4.  They were:
1. hasA
2. contains
3. affects
4. connectedTo

Eventually this style of hierarchy was abandoned, in part because of the explosion of inferred triples.

So, while it is no doubt true that a bunch of our properties could be correctly made into subproperties of affects/isAffectedBy, it would have two potential downsides.
- explosion of inferred triples
- it would be some extra work to decide which properties are sub properties. There would likely be gray areas.

One non-trivial upside is
-  a bit easier to browse the properties as a hierarchy, say in Protege.

So, I'm leaning towards skipping it.

**Comment by justin2004** [comment]
> explosion of inferred triples

i don't see that as a clear downside. more triples means more ways for triples to satisfy queries.

also this is a _huge_ upside:

> a bit easier to browse the properties as a hierarchy, say in Protege.

because if more rungs on the ladder means users of gist can walk down to the correct usage then that makes usages a gist more likely to converge.

**Comment by uscholdm** [comment]
> i don't see that as a clear downside. more triples means more ways for triples to satisfy queries.

We recently got rid of all property inverses because of severe performance issues due to doubling the numeber of triple for each propety that had an inverse. It was a real issue that drove this decision. The extra triples for subproperties is potentially much worse, depending on how many levels in the hierarchy.

**Comment by uscholdm** [comment]
> also [browsing the property hierarchy] is a _huge_ upside

Yes, it could be a big deal especially for new users of gist. Interestingly, removing inverses made browsing the hierarchy easier, since inverses showed up as entirely unrelated to each other in the hierarchy due to alpha sorting.
